pkg/scheduler/scheduler.go

```diff
@@ -73,16 +73,11 @@ func NewScheduler() *Scheduler {
 	return s
 }
 
-func (s *Scheduler) onUpdateNode(_, newObj interface{}) {
-	s.nodeNotify <- struct{}{}
-}
-
-func (s *Scheduler) onDelNode(obj interface{}) {
-	s.nodeNotify <- struct{}{}
-}
-
-func (s *Scheduler) onAddNode(obj interface{}) {
-	s.nodeNotify <- struct{}{}
+func (s *Scheduler) doNodeNotify() {
+	select {
+	case s.nodeNotify <- struct{}{}:
```

Comment:
Current nodeNotify channel length is 1, if we use select to write, maybe will lost event.

Reply:
Since RegisterFromNodeAnnotations actually handles all nodes, we just need to make sure that there is a signal in nodeNotify when the node changes, so that the node will not be ignored during a RegisterFromNodeAnnotations execution.

Reply:
After reading the RegisterFromNodeAnnotations code, I understand it, 
but I think we can create an issue to optimize the implementation of RegisterFromNodeAnnotations here; handle each node separately. no need to process all nodes every time.

Reply:
yeah, that would be a rather large modification, maybe we can fix this first, otherwise the number of nodes needed during the initialization phase of the program leads to a lot of duplicate RegisterFromNodeAnnotations executions